src/supertux/menu/addon_browse_menu.cpp

```diff
@@ -0,0 +1,240 @@
+//  SuperTux
+//  Copyright (C) 2022 Vankata453
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#include "supertux/menu/addon_browse_menu.hpp"
+
+#include <fmt/format.h>
+
+#include "addon/addon.hpp"
+#include "addon/addon_manager.hpp"
+#include "gui/dialog.hpp"
+#include "gui/menu_item.hpp"
+#include "gui/menu_manager.hpp"
+#include "supertux/menu/addon_preview_menu.hpp"
+#include "supertux/menu/download_dialog.hpp"
+#include "util/log.hpp"
+
+namespace {
+
+#define IS_REPOSITORY_MENU_ID(idx) (((idx) - MNID_ADDON_LIST_START) % 2 == 0)
+
+#define MAKE_REPOSITORY_MENU_ID(idx) (MNID_ADDON_LIST_START + 2 * (idx) + 0)
+
+#define UNPACK_REPOSITORY_MENU_ID(idx) ((((idx) - MNID_ADDON_LIST_START) - 0) / 2)
+
+} // namespace
+
+AddonBrowseMenu::AddonBrowseMenu(const bool langpacks_only, const bool auto_install_langpack) :
+  m_addon_manager(*AddonManager::current()),
+  m_repository_addons(),
+  m_langpacks_only(langpacks_only),
+  m_auto_install_langpack(auto_install_langpack),
+  m_browse_page(1),
+  m_max_addons_on_page(10)
+{
+  refresh();
+  if (auto_install_langpack)
+  {
+    const std::string& language = g_dictionary_manager->get_language().get_language();
+    if (language == "en")
+      return;
+    check_online();
```

Comment:
Check for English language shouldn't be necessary, and shouldn't be hardcoded here. (Some people may be using the game in English to install a langpack for their language)

Reply:
Since this does not affect normal usage of the add-on browsing menu (the user could always download anything they want, no matter the display language), but rather is used to automate the process of auto-installing the main language pack if the user chooses a language different than English, I will leave it here until further notice.